controlplane/pkg/common/request_builder.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2019 Cisco and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/networkservice"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/registry"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/model"
+)
+
+type RequestBuilder interface {
+	Build(string, *registry.NSERegistration, *model.Forwarder, *connection.Connection) *networkservice.NetworkServiceRequest
+}
+
+type LocalRequestBuilder struct {
+	nsmName     string
+	idGenerator func() string
+}
+
+type RemoteRequestBuilder struct {
+	nsmName     string
+	idGenerator func() string
+}
+
+func createLocalNSERequest(connectionId string, idGenerator func() string, nsmName string, localMechanisms []*connection.Mechanism, requestConn *connection.Connection) *networkservice.NetworkServiceRequest {
+	if connectionId == "" {
```

Comment:
Let's move the check for `connectionId` to method `Build`.
It will reduce args length for fucntion`createLocalNSERequest`.

Reply:
Therefore we need to duplicate `connectionId` check twice, because `createLocalNSERequest` is called twice